articles/architecture-scenarios/implementation/b2c/authentication.md

```diff
@@ -0,0 +1,148 @@
+
+::: warning
+Native mobile applications (and desktop applications) should use the system browser for authentication, or they open themselves up to additional security risks. See [Native vs. Browser Login on Mobile](/design/browser-based-vs-native-experience-on-mobile) for more information. 
+:::
+
+As discussed, we have found that most Auth0 customers use OpenID Connect (OIDC) as their industry-standard protocol when it comes to customer facing applications. Figuring out which [OIDC flow](/api-auth/intro) to use is your first task, and you will want to start by reviewing the [grant type mapping](/applications/reference/grant-types-auth0-mapping) guidance.  
+
+If you want to allow anonymous users access to any part of our application then you need to determine if you will be redirecting right away or prompting your users to redirect only when required (or perhaps some combination of both; see [Redirect Users After Login](/users/guides/redirect-users-after-login) for further discussion). If users can deep link to a protected version (or area) of your site then you will need to determine the links to your application that will result in an automatic redirect to Auth0. 
+
+### Anonymous access
+
+It is important to consider the user experience when someone first comes to the application. If your application supports anonymous user access (quite common for eCommerce applications) there are different scenarios to consider:
+
+* They are returning to the application after having already logged in, or
+* This is the first time they are accessing the application:
+  * They have already accessed a different application that uses the same Auth0 tenant,
+  * They have never (or not in a long time) authenticated on this device or browser.
+
+When an anonymous user accesses your application, it can often be desirable for the application to discover if the user has already logged into a different application in the same family, or to remember this user even if the application is a SPA with no state. For example, if you can determine that the user is already logged in, you might decide to have the header of the application skip displaying a login button and instead have an account or profile menu for the user. To accomplish this you will want to utilize "silent authentication". Silent authentication will allow you to check to see if the user is logged in without prompting them to log in if they are not. Then the application can present a login button if necessary. If the user is logged in already, however, then you will receive tokens and will not have to present the user with a login button again.
+
+::: warning
+Checking for a login session by redirecting to Auth0 can be really helpful for your application, but if this will result in a lot of requests it is important to employ some sort of throttling mechanism to avoid latency and/or rate limiting.
+<%= include('../../_includes/_rate-limit-policy.md') %>
```

Comment:
Might look better if we can include this as a continuation of the previous paragraph and not as a new paragraph?

Reply:
Fixed.